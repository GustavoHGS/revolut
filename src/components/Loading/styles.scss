@import "assets/_variables.scss";

.loading-container {
  overflow: show;
  position: fixed;
  z-index: 9999;
  height: 100%;
  width: 100%;
  margin: auto;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  padding-top: 12%;
  display: block;
}

.loading-background {
  background-color: #fff;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  border-radius: 50%;
  width: 100px;
  height: 100px;
  position: absolute;
  left: 48%;
  top: 40%;
  display: flex;
  background-color: $blue-dark;
}

.loading-background > div {
  cursor: default;
  margin-top: -8px !important;
}
