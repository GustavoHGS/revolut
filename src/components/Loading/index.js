import React from 'react'
import { connect } from 'react-redux'
import Lottie from 'react-lottie'
import PropTypes from 'prop-types'
import * as animation from 'assets/animations/loading.json'
import './styles.scss'

const defaultOptions = {
  loop: true,
  autoplay: true,
  animationData: animation,
  rendererSettings: {
    preserveAspectRatio: 'xMidYMid slice',
  },
}

const Loading = ({ isLoading }) => {
  if (!isLoading) {
    return null
  }
  return (
    <div className="loading-container">
      <div className="loading-background">
        <Lottie
          options={defaultOptions}
          height={120}
          width={120}
          isStopped={false}
          isPaused={false}
        />
      </div>
    </div>
  )
}

const mapStateToProps = state => ({
  isLoading: state.application.isApplicationLoading,
})

export default connect(mapStateToProps)(Loading)


Loading.propTypes = {
  isLoading: PropTypes.bool,
}

Loading.defaultProps = {
  isLoading: false,
}
