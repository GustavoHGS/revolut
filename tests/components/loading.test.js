import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import Loading from '../../src/components/Loading'

describe('Loading Component', () => {
  const initialState = { application: { isApplicationLoading: false } }
  const mockStore = configureStore()
  let store
  beforeEach(() => {
    store = mockStore(initialState)
  })

  it ('matches the loading snapshot', () => {
    const nodeTree = renderer.create(<Provider store={store}><Loading /></Provider>).toJSON()
    expect(nodeTree).toMatchSnapshot()
  })
})
