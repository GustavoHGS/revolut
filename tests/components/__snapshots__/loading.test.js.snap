// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component matches the loading snapshot 1`] = `null`;
